### PR TITLE
BUG-Fix: AssertionError when slicing MultiIndex and setting value of …

### LINKED ISSUE #34870: BUG: AssertionError when slicing MultiIndex and setting value of pandas.Series
#### Code Sample

```python
>>> import numpy as np
... import pandas as pd
... 
... arrays = [['bar', 'bar', 'baz', 'baz', 'foo', 'foo', 'qux', 'qux'],
...           ['one', 'two', 'one', 'two', 'one', 'two', 'one', 'two']]
... tuples = list(zip(*arrays))
... index = pd.MultiIndex.from_tuples(tuples, names=['first', 'second'])
... 
... s = pd.Series(np.random.randn(8), index=index)
>>>s
first  second
bar    one      -0.592576
       two      -0.054856
baz    one      -0.169856
       two      -0.712601
foo    one       2.180688
       two      -0.032646
qux    one      -0.729950
       two       0.883029
dtype: float64

>>> s[('baz', 'one'):('foo', 'two')] = 4  # This works!
>>> s
first  second
bar    one      -0.592576
       two      -0.054856
baz    one       4.000000
       two       4.000000
foo    one       4.000000
       two       4.000000
qux    one      -0.729950
       two       0.883029
dtype: float64

>>> s.loc[('baz', 'one'):('foo', 'two')] = 4  # This gives the error!
Traceback (most recent call last):
  File "/Users/edoardo/miniconda3/envs/my37/lib/python3.7/site-packages/pandas/core/indexes/base.py", line 2646, in get_loc
    return self._engine.get_loc(key)
  File "pandas/_libs/index.pyx", line 111, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/index.pyx", line 138, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/hashtable_class_helper.pxi", line 1619, in pandas._libs.hashtable.PyObjectHashTable.get_item
  File "pandas/_libs/hashtable_class_helper.pxi", line 1627, in pandas._libs.hashtable.PyObjectHashTable.get_item
KeyError: ('baz', 'one')
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/Users/edoardo/miniconda3/envs/my37/lib/python3.7/site-packages/pandas/core/indexes/multi.py", line 2890, in _get_level_indexer
    start = level_index.get_loc(key.start)
  File "/Users/edoardo/miniconda3/envs/my37/lib/python3.7/site-packages/pandas/core/indexes/base.py", line 2648, in get_loc
    return self._engine.get_loc(self._maybe_cast_indexer(key))
  File "pandas/_libs/index.pyx", line 111, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/index.pyx", line 138, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/hashtable_class_helper.pxi", line 1619, in pandas._libs.hashtable.PyObjectHashTable.get_item
  File "pandas/_libs/hashtable_class_helper.pxi", line 1627, in pandas._libs.hashtable.PyObjectHashTable.get_item
KeyError: ('baz', 'one')
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "/Users/edoardo/miniconda3/envs/my37/lib/python3.7/site-packages/pandas/core/indexing.py", line 670, in __setitem__
    indexer = self._get_setitem_indexer(key)
  File "/Users/edoardo/miniconda3/envs/my37/lib/python3.7/site-packages/pandas/core/indexing.py", line 641, in _get_setitem_indexer
    return ax.get_loc(key)
  File "/Users/edoardo/miniconda3/envs/my37/lib/python3.7/site-packages/pandas/core/indexes/multi.py", line 2662, in get_loc
    loc = self._get_level_indexer(key, level=0)
  File "/Users/edoardo/miniconda3/envs/my37/lib/python3.7/site-packages/pandas/core/indexes/multi.py", line 2903, in _get_level_indexer
    key.start, key.stop, key.step, kind="loc"
  File "/Users/edoardo/miniconda3/envs/my37/lib/python3.7/site-packages/pandas/core/indexes/base.py", line 4717, in slice_indexer
    raise AssertionError("Start slice bound is non-scalar")
AssertionError: Start slice bound is non-scalar

```

#### Problem description

I'm trying to slice and set values of a pandas Series but using the loc function does not work. I can do it by simply using [] and using loc if the Series is first converted into a DataFrame.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.7.7.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 19.5.0
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : None.UTF-8
pandas           : 1.0.4
numpy            : 1.18.1
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 20.1.1
setuptools       : 47.3.0.post20200616
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.5.0
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : None
pandas_datareader: None
bs4              : 4.9.0
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : 4.5.0
matplotlib       : 3.1.3
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
pytest           : None
pyxlsb           : None
s3fs             : None
scipy            : 1.4.1
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
xlsxwriter       : None
numba            : None

</details>

### PR DESCRIPTION
**Pull request by luckydenis**
- [x] closes #34870 
- [x] tests added / passed (1 / 1)
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by LuckyDenis** [comment]
@dsaxton, please, can you watch it. I looked in more detail. In variables, `start_slice` and `end_slice` are lists with coordinates. Therefore, we get an error when checking for a scalar. Having studied the code, I did not find the reasons why we cannot process them.

**Comment by dsaxton** [comment]
> @dsaxton, please, can you watch it. I looked in more detail. In variables, `start_slice` and `end_slice` are lists with coordinates. Therefore, we get an error when checking for a scalar. Having studied the code, I did not find the reasons why we cannot process them.

I'm not very familiar with this part of the code base so take my advice with a grain of salt, but as far as I can tell start_slice and end_slice _should_ be scalars so we don't want to just remove these assertions as you're doing here.

I would instead look upstream of this to figure out why this is happening (it does seem that the Index is mangled by the time we get here and is no longer a MultiIndex, if that's any clue).

**Comment by LuckyDenis** [comment]
@dsaxton, I rolled back the changes and took your advice into account. You can view?

File: pandas/tests/indexing/test_loc.py [hunk 1]
```diff
@@ -1119,3 +1119,22 @@ def test_loc_with_period_index_indexer():
...
+    answer = [1, 1, 100, 100, 100, 100, 1, 1]
+
+    tuples = list(zip(*arrays))
+    index = pd.MultiIndex.from_tuples(tuples, names=["first", "second"])
+
+    dt1 = pd.Series(array, index=index)
+    dt2 = pd.Series(answer, index=index)
+
+    dt1.loc[("baz", "one"):("foo", "two")] = 100
+    tm.assert_series_equal(dt1, dt2)
```
Comment (jreback) [inline pandas/tests/indexing/test_loc.py:1140]: can u do
result =
expected =

File: pandas/core/indexes/multi.py [hunk 2]
```diff
@@ -2929,12 +2929,17 @@ def convert_indexer(start, stop, step, indexer=indexer, codes=level_codes):
...
             except KeyError:
 
-                # we have a partial slice (like looking up a partial date
-                # string)
-                start = stop = level_index.slice_indexer(
-                    key.start, key.stop, key.step, kind="loc"
-                )
-                step = start.step
+                if isinstance(key.start, tuple) or isinstance(key.stop, tuple):
+                    return convert_indexer(key.start, key.stop, key.step)
```
Comment (jreback) [inline pandas/core/indexes/multi.py:2933]: rather than returning should not just do

start = key.start
stop = key.stop

and fall thru?

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by simonjayhawkins** [comment]
@LuckyDenis can you address comments and add a release note

File: pandas/core/indexes/multi.py (hunk 2 above)
Comment (LuckyDenis) [reply pandas/core/indexes/multi.py:2933]: I don't quite understand what you mean here. Could you please explain?

Comment (jreback) [reply pandas/core/indexes/multi.py:2933]: add those 2 lines, then you don't need this if else.

**Comment by simonjayhawkins** [comment]
@LuckyDenis can you address @jreback comment https://github.com/pandas-dev/pandas/pull/35018#discussion_r468680783

**Comment by dsaxton** [comment]
Closing as stale

**Pull request closed without merging**